### PR TITLE
[DM local testing] Docker image and tutorial for local testing 

### LINKED ISSUE #291: [DM local testing] Docker image and tutorial for local testing
This is a feature request to enable our users to run the bats tests easily with their local codebase.

Key features:
- A container image stored in GCR which can be ran with the path of the bats test as an input parameter
- Dockerfile and cloudbuild.yaml to build this image
- Ability to provide the gcloud credentials from the local system. ( Ideally automatically picking it up, at least in the cloud shell environment.)
- Easy way to provide the required properties for .cloud-foundation-tests.yaml
- Easy way to provide the source code from the local file system. ( The entire templates dir.)
- Readme file which explains how to use the container image locally with docker

This feature is separate from the container used for the CI system.

### PR DESCRIPTION
**Pull request by kopachevsky**
Additional scope implemented.

Fixed dm/CI/cft_base_container - wasn't able to build it initially.
Added make command to build cft-base-image to build it.
Added cft-test-bats command to run test locally, example command:
   cd ./dm
   make cft-test-bats TEST=templates/instance/tests/integration/instance.bats

Fixes #291

### CODE REVIEW

File: dm/CI/cft_base_contianer/Dockerfile [hunk 1]
```diff
@@ -1,24 +1,27 @@
 FROM gcr.io/cloud-builders/gcloud
 
-RUN apt-get update && apt-get -y install make \
+
+
```
Comment (ivankorn) [inline dm/CI/cft_base_contianer/Dockerfile]: @kopachevsky  do we need these empty lines?

File: dm/CI/cft_base_contianer/Dockerfile [hunk 2]
```diff
@@ -1,24 +1,27 @@
...
     && make template-prerequisites \
     && make cft-prerequisites \
     && . venv/bin/activate \
     && ./src/cftenv \
     && pwd \
     && cft --version \
-    && bats -v
+    && bats -v \
+    && which bats
+
```
Comment (ivankorn) [inline dm/CI/cft_base_contianer/Dockerfile]: Do we need this empty line?

File: dm/CI/cft_base_container/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    && make template-prerequisites \
+    && make cft-prerequisites \
+    && . venv/bin/activate \
+    && ./src/cftenv \
+    && pwd \
+    && cft --version \
+    && bats -v \
+    && which bats
+
+
```
Comment (ivankorn) [inline dm/CI/cft_base_container/Dockerfile]: @kopachevsky do we need this empty line?

File: dm/CI/cft_base_container/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,27 @@
+FROM gcr.io/cloud-builders/gcloud
+
+
```
Comment (ivankorn) [inline dm/CI/cft_base_container/Dockerfile]: @kopachevsky do we need these empty lines?

**Comment by ocsig** [comment]
One things to change, can you please remove the cft_base_contianer folder and make it a symbolic link to cft_base_container?

**Comment by kopachevsky** [comment]
@ocsig 

> One things to change, can you please remove the cft_base_contianer folder and make it a symbolic link to cft_base_container?

It's already there but github for some reason don't show simbolic link file here, checkout branch to check it.

**Pull request merged**